include/dr.h

```diff
@@ -27,9 +28,19 @@ typedef pvr_vertex_t plx_vertex_t;
 #define PLX_VERT	PVR_CMD_VERTEX
 #define PLX_VERT_EOS	PVR_CMD_VERTEX_EOL
 
-#define plx_dr_init(a) pvr_dr_init(*a)
-#define plx_dr_target(a) pvr_dr_target(*a)
-#define plx_dr_commit(a) pvr_dr_commit(a)
+static inline void plx_dr_init(plx_dr_state_t *state) {
+  pvr_dr_init((pvr_dr_state_t *)state);
+}
+
+static inline plx_vertex_t *plx_dr_target(plx_dr_state_t *state) {
+  return (plx_vertex_t *)pvr_dr_target(*(pvr_dr_state_t*)state);
+}
+
+static inline void plx_dr_commit(plx_vertex_t *vertex) {
+  pvr_dr_commit((pvr_vertex_t *)vertex);
+}
+
+#define plx_dr_finish pvr_dr_finish
```

Comment:
Why leave this one as a macro when the rest are functions?

Reply:
Yeah, good point. In my head it was because the parameter types weren't changed via typedef, but I might as well be consistent.